### ISSUE TITLE
Prompt validator can't be async when using bypass

### ISSUE THREAD

**Issue by ThisIsMissEm** [plopjs/node-plop#182]
Inquirer allows for the prompt's validate function to be asynchronous, however, when Plop prompt-bypass function calls the validator for the prompt, it does not allow for this and expects all validators to return a `boolean` or `string`.

### Example

```
  plop.setGenerator('user', {
    description: 'Add a new user',
    prompts: [
      {
        name: 'username',
        message: 'What is the user name?',
        type: 'input',
        validate: async (input) => {
          // do something async, returning either a boolean or string;
        },
      },
    ], 
    actions: []
  })
```

Calling with `plop user` works, but `plop user thisismissem` fails with `[ERROR] [object Promise]`

**Comment by amwmedia** [comment]
Agreed, this must have been an oversight when we built the bypass functionality.

**Comment by crutchcorn** [comment]
This issue is being fixed, with tests to verify, in `node-plop@0.13` and `plop@3`. Tracking PR: #212

**Comment by crutchcorn** [comment]
This has now been released as-of Plop 3!

https://github.com/plopjs/plop/releases/tag/v3.0.0
